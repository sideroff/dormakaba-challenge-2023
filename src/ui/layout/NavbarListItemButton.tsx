import { useRouter } from 'next/router';
import Link from 'next/link';
import ListItemButton from '@mui/material/ListItemButton';
import Typography from '@mui/material/Typography';

interface NavbarListItemButtonProps {
  label: string;
  href: string;
}

export function NavbarListItemButton({
  label,
  href,
}: NavbarListItemButtonProps) {
  const { pathname } = useRouter();

  return (
    <ListItemButton
      selected={pathname === href}
      sx={{ color: 'common.black', mr: 2, flexGrow: 'initial' }}
      LinkComponent={Link}
      href={href}
    >
      <Typography>{label}</Typography>
    </ListItemButton>
  );
}
