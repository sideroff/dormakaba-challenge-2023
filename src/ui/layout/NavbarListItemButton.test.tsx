import { render } from '@testing-library/react';
import mockRouter from 'next-router-mock';
import { NavbarListItemButton } from './NavbarListItemButton';

jest.mock('next/router', () => require('next-router-mock'));

describe('NavbarListItemButton', () => {
  it('should render correctly', () => {
    mockRouter.push('/test');

    const { container } = render(
      <NavbarListItemButton label="Test" href="/test" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
