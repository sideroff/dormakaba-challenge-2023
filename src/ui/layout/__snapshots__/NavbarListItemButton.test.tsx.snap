// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarListItemButton should render correctly 1`] = `
<a
  class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters Mui-selected MuiListItemButton-root MuiListItemButton-gutters Mui-selected css-72rlw3-MuiButtonBase-root-MuiListItemButton-root"
  href="/test"
  tabindex="0"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Test
  </p>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</a>
`;
